### PR TITLE
{data}[GCC/11.3.0] xmlf90 v1.5.6

### PR DESCRIPTION
**Pull request by ahkole**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/x/xmlf90/xmlf90-1.5.6-GCC-11.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+source_urls = ['https://gitlab.com/siesta-project/libraries/xmlf90/-/archive/%(version)s']
+sources = [SOURCELOWER_TAR_GZ]
+checksums = ['c4492596c911b668b94fee40a53453f06aef3bc41014b9e2f4b51e640ad20528']
+
+builddependencies = [
+    ('CMake', '3.23.1'),
+]
+
+separate_build_dir = True
+
```
Comment (Micket) [inline easybuild/easyconfigs/x/xmlf90/xmlf90-1.5.6-GCC-11.3.0.eb:20]:
```suggestion
```

This is the default, can be excluded

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
vera-c1 - Linux Rocky Linux 8.6, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/Micket/6e6abf2e77723e9fdd8cbe81403f49a5 for a full test report.

**Review by Micket — CHANGES_REQUESTED** [review]

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
vera-c1 - Linux Rocky Linux 8.6, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/Micket/edb6b15050f093716946ca3160e18a7e for a full test report.

**Comment by Micket** [comment]
I see the cmakemake build tool decides to put the mod files under
```
include/xmlf90/GNU-11.3.0/*.mod
```
as opposed to
```
include/*.mod
```
inthe previous versions.

Are libraries that depend on xmlf90 going to find these hidden mod files when they need it?

**Comment by ahkole** [comment]
> I see the cmakemake build tool decides to put the mod files under
> 
> ```
> include/xmlf90/GNU-11.3.0/*.mod
> ```
> 
> as opposed to
> 
> ```
> include/*.mod
> ```
> 
> inthe previous versions.
> 
> Are libraries that depend on xmlf90 going to find these hidden mod files when they need it?

They do have to update their build system but then they can find the include files. Projects that use `cmake` or `pkgconfig` to detect dependencies should be able to automatically set the include directory to find the include files (see the new `libPSML` library for which I also opened a PR which depends on this new version of `xmlf90`).

**Comment by Micket** [comment]
That extra layer subdir for the mod files is a thing i've seen in the past for fortran projects that want to support ABI for multiple compilers which has caused minor problems in the past where extra -I flags had to be specified for all dependent software. This subdir is unncessary due to the way easybuild installs each toolchain under a different prefix though, so i vaguely recall us getting for of that extra dir (patching it out) for some project.

But, at least libPSML seems to be happy to find it without any extra manual steps, so i guess it's all good like this.

**Review by Micket — APPROVED** [review]
lgtm

**Comment by Micket** [comment]
Going in, thanks @ahkole!

**Pull request merged**